apps/api/src/app/portfolio/calculator/roai/portfolio-calculator-btcusd.spec.ts

```diff
@@ -0,0 +1,246 @@
+        totalInvestmentValueWithCurrencyEffect: 44558.42,
+        value: 44558.42,
+        valueWithCurrencyEffect: 44558.42
+      });
+
+      expect(
+        portfolioSnapshot.historicalData[
+          portfolioSnapshot.historicalData.length - 1
+        ]
+      ).toEqual({
+        date: '2022-01-14',
+        investmentValueWithCurrencyEffect: 0,
+        netPerformance: -1463.18,
+        netPerformanceInPercentage: -0.032837340282712,
+        netPerformanceInPercentageWithCurrencyEffect: -0.032837340282712,
+        netPerformanceWithCurrencyEffect: -1463.18,
+        netWorth: 43099.7,
+        totalAccountBalance: 0,
+        totalInvestment: 44558.42,
+        totalInvestmentValueWithCurrencyEffect: 44558.42,
+        value: 43099.7,
+        valueWithCurrencyEffect: 43099.7
+      });
+
+      expect(portfolioSnapshot).toMatchObject({
+        currentValueInBaseCurrency: new Big('43099.7'),
+        errors: [],
+        hasErrors: false,
+        positions: [
+          {
+            averagePrice: new Big('44558.42'),
+            currency: 'USD',
+            dataSource: 'YAHOO',
+            dividend: new Big('0'),
+            dividendInBaseCurrency: new Big('0'),
+            fee: new Big('4.46'),
+            feeInBaseCurrency: new Big('4.46'),
+            firstBuyDate: '2021-12-12',
+            grossPerformance: new Big('-1458.72'),
+            grossPerformancePercentage: new Big('-0.03273724696701543726'),
+            grossPerformancePercentageWithCurrencyEffect: new Big(
+              '-0.03273724696701543726'
+            ),
+            grossPerformanceWithCurrencyEffect: new Big('-1458.72'),
+            investment: new Big('44558.42'),
+            investmentWithCurrencyEffect: new Big('44558.42'),
+            netPerformance: new Big('-1463.18'),
+            netPerformancePercentage: new Big('-0.03283734028271199921'),
+            netPerformancePercentageWithCurrencyEffectMap: {
+              max: new Big('-0.03283734028271199921')
+            },
+            netPerformanceWithCurrencyEffectMap: {
+              max: new Big('-1463.18')
+            },
+            marketPrice: 43099.7,
+            marketPriceInBaseCurrency: 43099.7,
+            quantity: new Big('1'),
+            symbol: 'BTCUSD',
+            tags: [],
+            timeWeightedInvestment: new Big('44558.42'),
+            timeWeightedInvestmentWithCurrencyEffect: new Big('44558.42'),
+            transactionCount: 1,
+            valueInBaseCurrency: new Big('43099.7')
+          }
+        ],
+        totalFeesWithCurrencyEffect: new Big('4.46'),
+        totalInterestWithCurrencyEffect: new Big('0'),
+        totalInvestment: new Big('44558.42'),
+        totalInvestmentWithCurrencyEffect: new Big('44558.42'),
+        totalLiabilitiesWithCurrencyEffect: new Big('0'),
+        totalValuablesWithCurrencyEffect: new Big('0')
+      });
+
+      expect(portfolioSnapshot.historicalData.at(-1)).toMatchObject(
```

Comment:
This part is currently missing in `portfolio-calculator-btceur.spec.ts`, unsure if needed.